core-bundle/src/Resources/contao/templates/frontend/fe_page.html5

```diff
@@ -13,6 +13,12 @@
       <meta name="generator" content="Contao Open Source CMS">
     <?php $this->endblock(); ?>
 
+    <?php $this->block('canonical'); ?>
```

Comment:
Why does it have to be a block? It seems very unlikely to me that someone needs to replace it with a custom implementation.

Reply:
The more blocks, the better. The `fe_page` is our fundamental template so it should be absolutely stuffed with blocks imho.

Reply:
Imho it is likely that this may need to be replaced. If, for example, you have another solution to add a canonical tag via `$GLOBALS['TL_HEAD']`, you can simply add

```php
<?php $this->block('canonical') ?>
<?php $this->endblock(); ?>
```

to your `fe_page` template, in order to disable the built in solution. Of course you could also overwrite the `$this->canonical` template variable via a `parseTemplate` hook instead.

Reply:
If someone implements a custom canonical logic, it is very likely that they will only overwrite `$this->canonical` and still use the `<link rel="canonical" href="<?= $this->canonical; ?>">` markup – because there is no other way to set the canonical URL! Therefore, the block is never needed. The if condition is enough.

Reply:
https://github.com/Toflar/contao/pull/10